### PR TITLE
more typing annotations

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: apypie/api.py [hunk 1]
```diff
@@ -182,7 +183,7 @@ def _load_apidoc(self):
         return api_doc
 
     def _retrieve_apidoc(self):
-        # type: () -> dict
+        # type: () -> Optional[dict]
```
Comment (ekohl) [inline apypie/api.py:186]: When is this None? This also implies we can write `None` to `apidoc_file`. If we detect `response` is `None`, should we raise an exception?

Comment (evgeni) [reply apypie/api.py]: you're right, dropped the `Optional` here.

**Comment by evgeni** [comment]
the existing failures are unrelated and fixes are prepared in https://github.com/Apipie/apypie/pull/121 and https://github.com/Apipie/apypie/pull/122

**Comment by ekohl** [comment]
Those are merged, can you rebase?

**Comment by evgeni** [comment]
done

**Review by ekohl — APPROVED** [review]

**Pull request merged**